### ISSUE TITLE
Unable to lint R files - `lintr` package missing

### ISSUE THREAD

**Issue by tonofshell** [github/super-linter#1510]
## Describe the bug

When linting a repo containing R files (R or R Markdown), the R linting step always fails, seemingly because the `lintr` package is missing.

## Expected behavior

Any R files in a repo are linted.

## Steps to Reproduce

Activate the Super-Linter action on a repo with R files in it. 

## Logs

```
2021-04-30 23:09:59 [ERROR]   Found errors in [R] linter!
2021-04-30 23:09:59 [ERROR]   Error code: 1. Command output:
------
# Bootstrapping renv 0.12.5 --------------------------------------------------
* Downloading renv 0.12.5 from CRAN archive ... OK
* Installing renv 0.12.5 ... Done!
* Successfully installed and loaded renv 0.12.5.
Error in loadNamespace(name) : there is no package called ‘lintr’
Calls: :: ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
------
2021-04-30 23:10:00 [INFO]   ----------------------------------------------
2021-04-30 23:10:00 [INFO]   ----------------------------------------------
2021-04-30 23:10:00 [INFO]   The script has completed
2021-04-30 23:10:00 [INFO]   ----------------------------------------------
2021-04-30 23:10:00 [INFO]   ----------------------------------------------
2021-04-30 23:10:01 [ERROR]   ERRORS FOUND in R:[1]
2021-04-30 23:10:02 [FATAL]   Exiting with errors found!
```

**Comment by admiralAwkbar** [comment]
@tonofshell seems to have been a pathing issue... pushing a release today

**Comment by admiralAwkbar** [comment]
@tonofshell can you validate with 3.16.3 release?

**Comment by tonofshell** [comment]
@admiralAwkbar Unfortunately, the issue is not resolved for me on 3.16.3. I got the exact same error as on 3.16.2.

**Comment by admiralAwkbar** [comment]
@tonofshell this is frustrating lol... I see the issue and it seems docker is not copying the files from the base image...
I'm fighting with it to see what I'm doing so wrong... Seems this was introduced when we starting multistage builds...
Sorry for the issues. trying to resolve it

**Comment by admiralAwkbar** [comment]
@tonofshell can you validate agin? or send some logging?
Im able to get it to work on `3.16.3`
```bash
$ docker run -it --entrypoint /bin/bash github/super-linter:v3.16.3
Unable to find image 'github/super-linter:v3.16.3' locally
v3.16.3: Pulling from github/super-linter
540db60ca938: Already exists
193cc745113a: Pull complete
c44012a38544: Pull complete
f5d900d43fde: Pull complete
3d2c88262125: Pull complete
23f640e0bfb5: Pull complete
6dc3b3acf8f9: Pull complete
2f70e83636a9: Pull complete
0f4cde0db2ae: Pull complete
868a7b61a8b3: Pull complete
8ea318640ea5: Pull complete
cadde9af1932: Pull complete
090d14f1fc99: Pull complete
955ec4a90ffb: Pull complete
444062ec19ca: Pull complete
5c0106cb842f: Pull complete
Digest: sha256:cf5b29d016aeb9df336e941a843aa20a268f27cf3ccd7ee1d183ea8591d11095
Status: Downloaded newer image for github/super-linter:v3.16.3
bash-5.1# vi testing/here/file.r
bash-5.1# R --slave -e "lints <- lintr::lint('testing/here/file.r');print(lints);errors <- purrr::keep(lints, ~ .\$type == 'error');quit(save = 'no', status = if (length(errors) > 0) 1 else 0)" 2>&1
/testing/here/file.r:49:1: style: Trailing blank lines are superfluous.

^
bash-5.1#
```

**Comment by tonofshell** [comment]
@admiralAwkbar Okay, this is even more puzzling. I can confirm there are no issues running directly in Docker. I created a new repo to try to reproduce the issue, but it also worked fine. Went back to the existing repo and re-ran the job and I'm getting the issue still. I used the exact same YAML for both and can confirm that both were pulling the 3.16.3 container. What additional logging can I send that would be helpful?

**Comment by admiralAwkbar** [comment]
will do some more testing. my main concern was a pull of the old image
Still hunting to see if maybe perms? or linkage?

**Comment by tonofshell** [comment]
@admiralAwkbar Okay, let me know if there's any additional information I can provide that would help. Happy to test anything as well!

**Comment by admiralAwkbar** [comment]
@tonofshell We upgraded to 3.17.0, has that helped?

**Comment by tonofshell** [comment]
@admiralAwkbar No luck, still does not work in the repos it was not working in before. I still have not found any pattern to why it works in some repos and not others.

**Comment by tonofshell** [comment]
@admiralAwkbar [I was able to reproduce the issue](https://github.com/n3-initiative/super-linter-testing/pull/1). If `renv` is activated in that branch, Super-Linter tries to load `lintr` from the `renv` environment and then can't find the package. However, it seems even if I load `lintr` in an R file so it gets into the `renv` environment, Super-Linter still fails to locate `lintr`.

**Comment by tonofshell** [comment]
@admiralAwkbar Just wanted to bump you on this, do you have any updates? Thanks!

**Comment by admiralAwkbar** [comment]
@tonofshell Hey!
sorry got super busy with work and now its on my list as it broke my new work :) so this is top priority now :)

**Comment by tonofshell** [comment]
@admiralAwkbar Ah, the circle of software development... No worries though, let me know if there's any other info/testing I can provide to help!

**Comment by admiralAwkbar** [comment]
@tonofshell this was fixed in the `4.0.0` release!

Sorry for the missed ping

**Comment by tonofshell** [comment]
@admiralAwkbar I'm still having the same issue on v4...
